agent/proxycfg/state.go

```diff
@@ -106,8 +107,10 @@ func copyProxyConfig(ns *structs.NodeService) (structs.ConnectProxyConfig, error
 // The returned state needs its required dependencies to be set before Watch
 // can be called.
 func newState(ns *structs.NodeService, token string) (*state, error) {
-	if ns.Kind != structs.ServiceKindConnectProxy && ns.Kind != structs.ServiceKindMeshGateway {
-		return nil, errors.New("not a connect-proxy or mesh-gateway")
+	if ns.Kind != structs.ServiceKindConnectProxy &&
```

Comment:
Maybe a switch statement would be prettier?
```
switch ns.Kind {
  case structs.ServiceKindConnectProxy:
  case structs.ServiceKindMeshGateway:
  case structs.ServiceKindIngressGateway:
  default:
    return nil, ERROR
}
```

Reply:
Seems reasonable to me, done!